docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+  - EJS has a large community of users which will ensure faster help and support.
+  - It is a simple templating engine and far easier to use as its plain Javascript and since we want to keep our mail templates straightforward, using EJS could be a good choice.
+- Cons
+  - Adds a new dependency to the stack
+  - Lacks support for `block` functionality, which will make resuing components a bit difficult.
+  - Although we can implement `block` functionality with the help of third-party support, most of the projects are outdated and not maintained anymore.
+
+### Pros and Cons of using MJML as part of our templating stack
+
+As part of modernizing the templates, we aim to replace the framework in which our templates are currently built in so as to include more logic in them while development. Along with the framework, we considered using [MJML](https://mjml.io/) as our potential markup language, designed to reduce the pain of coding a responsive email.
+
+- Pros
+  - MJML is a component-based library having its own pre-built components as well as gives us the ability to define custom compononents which translates to responsive HTML after compilation. These components can be reused across the templates and modified by passing parameters e.g. [this social links component](https://documentation.mjml.io/#mj-social)
+  - MJML can help us create responsive emails using mobile-first approach and it controls that responsiveness automatically needing very less to no developer intervention.
+  - This library has been created by keeping in mind all the industry standards as well as the quirks of old email clients, so we will eventually be saving time on investigating cross-browser compatability of our templates
+  - MJML has by far the highest weekly downloads and github stars in comparision to its alternatives like [Foundation for Emails](https://github.com/foundation/foundation-emails).
+  - MJML can be integrated with React to generate responsive emails and there is also a [library](https://github.com/wix-incubator/mjml-react) that could help us with that purpose.
+- Cons
+  - Adds a new dependency to the stack
+  - Creates a slight learning curve
+  - While we may not have much component reuse between other packages that use React anyway, using MJML in email components would make it more challenging to reuse them elsewhere
+
+## Pros and Cons of the Styling Options
+
+When emails are built by the server, we need the styles to be inserted inline since most email clients disregard style tags. However, inline styles aren't great for consistency across the look and feel of our emails and they make maintenance more difficult. We want to pick a solution that lets us use reusable classes during development but when built, can be turned into inline styles
+
+### Option A - Continue to use adhoc inline styling
+
+- Description
+  - Keep the current inline styling as is in the templates.
+- Pros
```

Comment:
A system like MJML drastically cuts down the CSS needed, as MJML components already encompass the complex CSS needed to render correctly in email clients. Only minimal tweaks are needed as inline CSS.

Perhaps worth adding as a separate Pro, is that having minimal inline CSS with MJML means one can safely and consistently change individual emails without worrying about how many other emails they might've broken if they were instead editing a shared CSS sheet with a system like Tailwind.

Reply:
In some sense, all of these options boil down to Option A or Option D. That is, we either:

* have manually maintained inline styles
* or we add a build step to post-process Tailwind / SASS / CSS-in-JS into inline styles or an embedded stylesheet in every email

For Tailwind, there's also [Maizzle](https://maizzle.com/) - which I've never used, but which claims to do the needful here.

MJML looks like a more drastic transformation, since it's more than just a CSS processor.

Reply:
Maizzle looks slick, 👍🏻